lib/android-helpers.js

```diff
@@ -628,16 +628,22 @@ helpers.verifyUnlock = async function (adb) {
 };
 
 helpers.initDevice = async function (adb, opts) {
-  await adb.waitForDevice();
-  // pushSettingsApp required before calling ensureDeviceLocale for API Level 24+
-  await helpers.pushSettingsApp(adb);
-  if (!opts.avd) {
-    await helpers.setMockLocationApp(adb, SETTINGS_HELPER_PKG_ID);
-  }
+  if (opts.skipDeviceInitialization) {
+    logger.info(`'skipDeviceInitialization' is set. Skipping device initialization.`);
+  } else {
+    await adb.waitForDevice();
+
+    // pushSettingsApp required before calling ensureDeviceLocale for API Level 24+
+    await helpers.pushSettingsApp(adb);
+    if (!opts.avd) {
+      await helpers.setMockLocationApp(adb, SETTINGS_HELPER_PKG_ID);
+    }
 
-  if (opts.language || opts.locale) {
-    await helpers.ensureDeviceLocale(adb, opts.language, opts.locale, opts.localeScript);
+    if (opts.language || opts.locale) {
```

Comment:
locale setting can also be preserved. It is only changed if `language` or `locale` options are present

Reply:
Actually, now I see where I was wrong. I preserved all actions dependent on other caps. They can be easily skipped anyway, if they are not needed. Sorry for that. I'm just starting jumping into that project from the code side.

Reply:
np, we have code review procedure exactly for this purpose